controller/public/javascripts/spectaql.min.js

```diff
@@ -0,0 +1 @@
+"use strict";function scrollSpy(){var l=5,e=document.querySelector("html"),r=(!e||(e=window.getComputedStyle(e).scrollPaddingTop)&&"string"==typeof e&&"auto"!==e&&e.endsWith("px")&&(l+=parseInt(e.split("px")[0])),"nav-scroll-active"),i="nav-scroll-expand",d=null,s=[];function t(){d=null;var e=document.querySelectorAll("[data-traverse-target]");Array.prototype.forEach.call(e,function(e){s.push({id:e.id,top:e.offsetTop})})}var n=debounce(function(){t(),o()},500),o=debounce(function(){var e,t,n,o,c=function(e){for(var t=e+l,n=0;n<s.length;n++){var o=s[n],c=s[n+1];if(t>=o.top&&(!c||t<c.top))return n}return-1}(document.documentElement.scrollTop||document.body.scrollTop);c!==d&&(c=s[d=c],o=function(e){return e&&e.closest?e.closest(".nav-group-section"):null},e=document.querySelector("."+r),t=o(c=c?document.querySelector("#nav a[href*="+c.id+"]"):null),n=(o=o(e))!==t,o&&n&&o.classList.remove(i),t&&n&&t.classList.add(i),c&&(c.classList.add(r),c.scrollIntoViewIfNeeded?c.scrollIntoViewIfNeeded():!c.scrollIntoView||0<=(o=(o=c).getBoundingClientRect()).top&&0<=o.left&&o.bottom<=(window.innerHeight||document.documentElement.clientHeight)&&o.right<=(window.innerWidth||document.documentElement.clientWidth)||c.scrollIntoView({block:"center",inline:"start"})),e&&e.classList.remove(r))},100);setTimeout(function(){t(),o(),window.addEventListener("scroll",o),window.addEventListener("resize",n)},300)}function toggleMenu(){var t="drawer-open",e=document.querySelector("#spectaql .sidebar-open-button"),n=document.querySelector("#spectaql #sidebar .close-button"),o=document.querySelector("#spectaql .drawer-overlay");function c(){var e=document.querySelector("#spectaql #page");e.classList.contains(t)?e.classList.remove(t):e.classList.add(t)}e.addEventListener("click",c),n.addEventListener("click",c),o.addEventListener("click",c)}function debounce(e,t){var n=null;return function(){clearTimeout(n),n=setTimeout(function(){e.apply(null)},t)}}window.addEventListener("DOMContentLoaded",e=>{toggleMenu(),scrollSpy()});
```

Comment:
Shouldn't this kind of dependency be installed through NPM ?

https://www.npmjs.com/package/spectaql